pkg/auth/authproviders/identity.go

```diff
@@ -32,6 +32,13 @@ func CreateRoleBasedIdentity(ctx context.Context, provider Provider, authResp *A
 		Attributes: authResp.Claims.Attributes,
 	}
 
+	// The attribute checker is not required for every provider, only check if set.
+	if provider.AttributeChecker() != nil {
+		if err := provider.AttributeChecker().Check(ud); err != nil {
```

Comment:
This checks `Rox` claims, i.e., attributes and their values we add to _our_ tokens while I thought we want to check claims in the IdP's tokens. I understand that the latter is tricky to implement in a uniform way so we should make sure we explain we talk about attributes "known" to us.

Reply:
I get the sentiment, I've tried to address this by making more comments, ensuring with a reference to tokens.RoxClaims, that we only take those into account and not IdP specific claims.